app/src/gui/components/record/form.tsx

```diff
@@ -957,26 +979,34 @@ class RecordForm extends React.Component<any, RecordFormState> {
           // update the component state with the new revision id and notify the parent
           try {
             this.setState({revision_cached: revision_id});
-            // SC. Removing this call since it prevents deletion of the draft
-            // later on (draftState.clear())
-            // by changing the draft state to 'uninitialised'
-            //
-            //this.formChanged(true, revision_id);
             if (this.props.setRevision_id !== undefined)
               this.props.setRevision_id(revision_id); //pass the revision id back
           } catch (error) {
             logError(error);
           }
-          return is_close === 'close'
-            ? (doc.data['hrid' + this.state.type_cached] ??
-                this.props.record_id)
-            : revision_id; // return revision id for save and continue function
+          // we need both the revision id and the record id below
+          return {
+            revision_id,
+            record_id: this.props.record_id,
+          } as RecordIdentifiers;
         })
         // generate a success alert
-        .then(hrid => {
+        .then((ids: RecordIdentifiers) => {
```

Comment:
thanks for actually typing this - always was insane that it said 'hrid' - it wasn't the HRID!?